### PR TITLE
DOC: Remove dead link and update links to https

### PR DESCRIPTION
**Pull request by mroeschke**
Removed one dead tutorial link and updated our doc links to https where available.

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @mroeschke

**Review by gfyoung — COMMENTED** [review]
https://github.com/pandas-dev/pandas/search?q=http%3A

@mroeschke : Were you trying to remove all of them, or just a subset?

File: doc/source/ecosystem.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ substantial projects that you feel should be on this list, please let us know.
 Statistics and Machine Learning
 -------------------------------
 
-`Statsmodels <http://www.statsmodels.org/>`__
+`Statsmodels <https://www.statsmodels.org/>`__
```
Comment (jreback) [inline doc/source/ecosystem.rst]: i think you have to make this the same length

File: doc/source/ecosystem.rst [hunk 2]
```diff
@@ -62,7 +62,7 @@ simplicity produces beautiful and effective visualizations with a
 minimal amount of code. Altair works with Pandas DataFrames.
 
 
-`Bokeh <http://bokeh.pydata.org>`__
+`Bokeh <https://bokeh.pydata.org>`__
```
Comment (jreback) [inline doc/source/ecosystem.rst]: same and so on

**Review by jreback — APPROVED** [review]
minor issues. merge on green.

**Comment by mroeschke** [comment]
@gfyoung just a subset, some of the links don't support https yet...

File: doc/source/ecosystem.rst (hunk 1 above)
Comment (mroeschke) [reply doc/source/ecosystem.rst]: Not sure what you mean. These are rendering okay and hyperlink to the correct page when I run it locally.

screen shot 2018-11-04 at 9 11 04 am

Comment (jreback) [reply doc/source/ecosystem.rst]: they are doc warnings on building i think

Comment (mroeschke) [reply doc/source/ecosystem.rst]: Gotcha. I see them. Will fix and push on green

**Pull request merged**